app/services/current-session.js

```diff
@@ -1,60 +1,43 @@
 import Service from '@ember/service';
 import { inject as service } from '@ember/service';
-import { get, computed } from '@ember/object';
-import { task, waitForProperty } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
 
-export default Service.extend({
-  session: service('session'),
-  store: service('store'),
-  async load() {
-    if (this.get('session.isAuthenticated')) {
-      const session = this.session;
-      const account = await this.store.find('account', get(session, 'data.authenticated.relationships.account.data.id'));
-      const user = await account.get('gebruiker');
-      const group = await this.store.find('bestuurseenheid', get(session, 'data.authenticated.relationships.group.data.id'));
-      const roles = await get(session, 'data.authenticated.data.attributes.roles');
-      this.set('_account', account);
-      this.set('_user', user);
-      this.set('_roles', roles);
-      this.set('_group', group);
+export default class CurrentSessionService extends Service {
+  @service session;
+  @service store;
+  @tracked account;
+  @tracked user;
+  @tracked roles;
+  @tracked group;
 
-      // The naming is off, but account,user,roles are taken for the
-      // promises in a currently public API.
-      this.setProperties({
-        accountContent: account,
-        userContent: user,
-        rolesContent: roles,
-        groupContent: group
-      });
+  async load() {
+    if (this.session.isAuthenticated) {
+      this.account = await this.store.find(
+        'account',
+        this.session.data.authenticated.relationships.account.data.id
+      );
+      this.user = await this.account.get('gebruiker');
```

Comment:
`get` shouldn't be needed if you await it.
```suggestion
      this.user = await this.account.gebruiker;
```